internal/alb/lb/attributes.go

```diff
@@ -0,0 +1,204 @@
+			}
+		case AccessLogsS3BucketString:
+			a.AccessLogsS3Bucket = aws.StringValue(attr.Value)
+		case AccessLogsS3PrefixString:
+			a.AccessLogsS3Prefix = aws.StringValue(attr.Value)
+		case IdleTimeoutTimeoutSecondsString:
+			a.IdleTimeoutTimeoutSeconds, err = strconv.ParseInt(aws.StringValue(attr.Value), 10, 64)
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+			if a.IdleTimeoutTimeoutSeconds < 1 || a.IdleTimeoutTimeoutSeconds > 4000 {
+				return a, fmt.Errorf("%s must be within 1-4000 seconds", aws.StringValue(attr.Key))
+			}
+		case RoutingHTTP2EnabledString:
+			a.RoutingHTTP2Enabled, err = strconv.ParseBool(aws.StringValue(attr.Value))
+			if err != nil {
+				return a, fmt.Errorf("invalid load balancer attribute value %s=%s", aws.StringValue(attr.Key), aws.StringValue(attr.Value))
+			}
+		default:
+			return a, fmt.Errorf("invalid load balancer attribute %s", aws.StringValue(attr.Key))
+		}
+	}
+	return a, nil
+}
+
+// AttributesController provides functionality to manage Attributes
+type AttributesController interface {
+	// Reconcile ensures the load balancer attributes in AWS matches the state specified by the ingress configuration.
+	Reconcile(context.Context, *Attributes) error
+}
+
+// NewAttributesController constructs a new attributes controller
+func NewAttributesController(elbv2 elbv2iface.ELBV2API) AttributesController {
+	return &attributesController{
+		elbv2: elbv2,
+	}
+}
+
+type attributesController struct {
+	elbv2 elbv2iface.ELBV2API
+}
+
+func (c *attributesController) Reconcile(ctx context.Context, desired *Attributes) error {
+	raw, err := c.elbv2.DescribeLoadBalancerAttributes(&elbv2.DescribeLoadBalancerAttributesInput{
+		LoadBalancerArn: aws.String(desired.LbArn),
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to retrieve attributes from ELBV2 in AWS: %s", err.Error())
+	}
+
+	current, _ := NewAttributes(raw.Attributes)
+
+	changeSet, ok := attributesChangeSet(current, desired)
+	if ok {
+		albctx.GetLogger(ctx).Infof("Modifying ELBV2 attributes to %v.", log.Prettify(changeSet))
+		_, err = c.elbv2.ModifyLoadBalancerAttributes(&elbv2.ModifyLoadBalancerAttributesInput{
+			LoadBalancerArn: aws.String(desired.LbArn),
+			Attributes:      changeSet,
+		})
+		if err != nil {
+			eventf, ok := albctx.GetEventf(ctx)
+			if ok {
+				eventf(api.EventTypeWarning, "ERROR", "%s attributes modification failed: %s", desired.LbArn, err.Error())
+			}
+			return fmt.Errorf("failed modifying attributes: %s", err)
+		}
+
+	}
+	return nil
+}
+
+// attributesChangeSet returns a list of elbv2.LoadBalancerAttribute required to change a into b
+func attributesChangeSet(a, b *Attributes) (changeSet []*elbv2.LoadBalancerAttribute, ok bool) {
+	if a.DeletionProtectionEnabled != b.DeletionProtectionEnabled && b.DeletionProtectionEnabled != DeletionProtectionEnabled {
```

Comment:
is the default compare desired?
I mean if user  changed b.DeletionProtectionEnabled  to true(via anno), and then change it back to false.
The a. DeletionProtectionEnabled(on alb) will always stays in true.

If we want to model the concept of optional attribute(like unset), we should use pointers to better model it. e.g. DeletionProtectionEnabled: *bool, which is why aws using *string instead of string. Hope golang got optional[bool] :D